posit-dev/positron
# ci: improved tags in pull-request

### Summary
Previously, re-running the `test-pull-request.yml` workflow failed to detect updated tags in the PR description because it relied on outdated data from the prior GitHub event. This issue has been resolved by using the GitHub API to fetch the latest PR information directly.

In addition, a comment is now posted to the PR to confirm the detected tags and provide feedback whenever tags are updated in the PR description.

### QA Notes
Confirmed I can re-run jobs after changing tags in PR description and it works as expected.

**Feel free to edit/add tags and see the comment update on this PR.**
@:data-explorer

### Screenshots
<img width="939" alt="Screenshot 2024-12-12 at 7 08 18 AM" src="https://github.com/user-attachments/assets/e506da57-294e-4a80-a6a5-6707fc37e5a6" />
Note: The `?` in the comment links to the new section I added in the README. It won't work right now since that readme section is not available in `main` yet.
